src/unstract/sdk/embedding.py

```diff
@@ -71,6 +71,7 @@ def _get_embedding(self) -> BaseEmbedding:
             ][Common.ADAPTER]
             embedding_metadata = embedding_config_data.get(Common.ADAPTER_METADATA)
             embedding_adapter_class = embedding_adapter(embedding_metadata)
+            self._usage_kwargs["provider"] = embedding_adapter_class.get_provider()
```

Comment:
@Deepak-Kesavan I don't remember how we use this usage dictionary - but if we try to generically fetch/ use it for all adapters it'll be good to add the same key for other adapters too. This will allow us to not worry about `KeyError`s while accessing it.
Ignore this comment if we explicitly read this `usage_kwargs` only for embedding and LLM types

Reply:
I see that we initialize it [here](https://github.com/Zipstack/unstract-sdk/pull/75/files#diff-ae0b5825b9cfb1f4a712504607c8101ae4f9c639a8b1ba08713c5558ba21b36b), ignore this comment